.github/workflows/build-and-test.yaml

```
@@ -12,22 +12,30 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - name: Checkout
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+        uses: actions/checkout@v4
       - name: Install dependencies
-        run: python3 -m pip install tox
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          source $HOME/.cargo/env
+          uv venv && source .venv/bin/activate
+          uv pip install -r requirements-dev.txt
       - name: Run linters
-        run: tox -e lint
+        run: uv run tox -e lint
 
   unit-test:
     name: Unit tests
     runs-on: ubuntu-22.04
     steps:
       - name: Checkout
-        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+        uses: actions/checkout@v4
       - name: Install dependencies
-        run: python -m pip install tox
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          source $HOME/.cargo/env
```

```suggestion
          sudo snap install --classic astral-uv
```

The snap's stable now :-)

---

Bit outta date though ;-)

---

Are you seeing a different version from what I am? It should have been 0.4.6 on Friday and is 0.4.7 now.

---

Latest version they've released is [0.6.4](https://github.com/astral-sh/ruff/releases/tag/0.6.4)